src/main/java/com/volmit/iris/core/TreeManager.java

```diff
@@ -0,0 +1,217 @@
+
+        // TODO: Implement placer
+        /*
+        IObjectPlacer placer = new IObjectPlacer(){
+
+            @Override
+            public int getHighest(int x, int z) {
+                return 0;
+            }
+
+            @Override
+            public int getHighest(int x, int z, boolean ignoreFluid) {
+                return 0;
+            }
+
+            @Override
+            public void set(int x, int y, int z, BlockData d) {
+
+            }
+
+            @Override
+            public BlockData get(int x, int y, int z) {
+                return null;
+            }
+
+            @Override
+            public boolean isPreventingDecay() {
+                return false;
+            }
+
+            @Override
+            public boolean isSolid(int x, int y, int z) {
+                return false;
+            }
+
+            @Override
+            public boolean isUnderwater(int x, int z) {
+                return false;
+            }
+
+            @Override
+            public int getFluidHeight() {
+                return 0;
+            }
+
+            @Override
+            public boolean isDebugSmartBore() {
+                return false;
+            }
+
+            @Override
+            public void setTile(int xx, int yy, int zz, TileData<? extends TileState> tile) {
+
+            }
+        };
+        */
+
+        // TODO: Figure out how to place without wrecking claims, other builds, etc.
+        // Especially with large object
+
+        // Place the object with the placer
+        /*
+        f.place(
+                event.getLocation() // TODO: Place the object at the right location (one of the center positions)
+                placer,
+                placement,
+                RNG.r,
+                Objects.requireNonNull(IrisWorlds.access(event.getWorld())).getData()
+        );
+        */
+        f.place(event.getLocation());
+    }
+
+    /**
+     * Finds a single object placement (which may contain more than one object) for the requirements species, location & size
+     * @param worldAccess The world to access (check for biome, region, dimension, etc)
+     * @param location The location of the growth event (For biome/region finding)
+     * @param type The bukkit TreeType to match
+     * @param size The size of the sapling area
+     * @return An object placement which contains the matched tree, or null if none were found / it's disabled.
+     */
+    private IrisObjectPlacement getObjectPlacement(IrisAccess worldAccess, Location location, IrisTreeType type, IrisTreeSize size) {
+
+        KList<IrisObjectPlacement> placements = new KList<>();
+
+        // Retrieve objectPlacements of type `species` from biome
+        IrisBiome biome = worldAccess.getBiome(location.getBlockX(), location.getBlockY(), location.getBlockZ());
+        placements.addAll(matchObjectPlacements(biome.getObjects(), size, type));
+
```

Comment:
You are using `worldAccess.getCompound().getRootDimension()`. This is incorrect. Instead you need to get the dimension at the given Y position.

```java
IrisDimension theRealDimension = ((Engine)worldAccess.getEngineAccess(yPosition)).getDimension();
```